src/matUtils/summary.cpp

```diff
@@ -9,12 +9,18 @@ po::variables_map parse_summary_command(po::parsed_options parsed) {
     conv_desc.add_options()
         ("input-mat,i", po::value<std::string>()->required(),
          "Input mutation-annotated tree file [REQUIRED]. If only this argument is set, print the count of samples and nodes in the tree.")
+        ("output-directory,d", po::value<std::string>()->default_value("./"),
+        "Write output files to the target directory. Default is current directory.")
         ("samples,s", po::value<std::string>()->default_value(""),
         "Write a tsv listing all samples in the tree and their parsimony score (terminal branch length).")
         ("clades,c", po::value<std::string>()->default_value(""),
         "Write a tsv listing all clades and the count of associated samples.")
         ("mutations,m", po::value<std::string>()->default_value(""),
         "Write a tsv listing all mutations in the tree and their occurrence count.")
+        ("aberrant,a", po::value<std::string>()->default_value(""),

```

Comment:
This was a feature request from Angie specifically, with an eye towards the future and providing tools to help track down malformations or potential problems with the MAT which may not prevent saving/loading/standard procedures but create unexpected behavior in some cases. She specifically requested checking for duplicate sample IDs and I added checking internal branch lengths and mutations after our discussion of unexpected behavior with path creation. For the most part I expect this to not be relevant, especially if people stay within the bounds of Usher and the tools we've written, but its very possible to directly edit a protobuf file via google's protoc compilers and it may be useful for debugging in the future.

Do you think I should change the conditions we check for, remove some, add some? Or do you disagree that its worth having at this time?

Reply:
As for internal node IDs, the ones I have here are overwritten with integers when the protobuf is saved and loaded in any case (you don't store internal node names in the Newick file). The new node identifiers never end up in any protobuf files because of this.

These identifiers only appear in output if Newick format is requested with the -R flag. In that use case, which is where I think people who want resolved trees will go, I thought it would be useful to have some way to identify the nodes which were altered by the resolution process instead of them being anonymous numbers.

This also gives us a straightforward way to verify what and where polytomies are being resolved or not for debugging purposes.

Reply:
If we write converters from other mutation-annotated tree formats (like Nextstrain/Auspice JSON) into protobuf then we will need to be more cautious about checking for things like no duplicate names.  TBH I don't think Nextstrain/Auspice tolerates duplicate names either -- but for general testing and debugging, it's good to have a sanity-checker to make sure that a new feature hasn't broken something that used to work, or that a bug report is not a garbage-in-garbage-out problem.

Reply:
Oh ok but the mutation_annotated_tree should throw an error already in creating nodes with duplicate names, so this might be a bit redundant for MAT files at least.